tests/publish.bats

```diff
@@ -0,0 +1,100 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test `flox publish` command
+#
+# This subcommand requires the configuration of a `cache_url`, i.e. a store to
+# copy binaries to, as well as a `sign_key` file to enfore signing binaries.
+# Both values can either be given via the command line as
+#     --cache-url <url>     and
+#     --sign-key <path>
+# or via the config subsystem.
+# Here, we use the config subsystem though the `FLOX_CACHE_URL`
+# and `FLOX_SIGN_KEY`env variables.
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# * Setup a nix substituter and configure flox to use it
+# * Get version and output hash for the `hello` package
+#   to check the generated catalog against.
+setup_file() {
+    nix-serve -p 8081 &
+    export NIX_SERVE_PID="$!"
+
+    # Set the `cache_url` config value
+    export FLOX_CACHE_URL="http://localhost:8081"
+
+    HELLO_OUT_PATH="$($FLOX_CLI nix eval --raw nixpkgs-flox#hello)"
+    export HELLO_HASH_FIRST_8="${HELLO_OUT_PATH:11:8}" # skip /nix/store, take 8
+    export HELLO_VERSION="$($FLOX_CLI nix eval --raw nixpkgs-flox#hello.version)"
+}
+
+# Giving each test an individual channel to allow parallel runs.
+# The setup will create a channel repo with a `hello` package.
+# `flox publish` resolves local repositories to their upstream counterpart.
+# Since we don't want to manage and pollute upstream repositories,
+# we set the repository as its own remote, and can verify publishes by checking
+# out the respective catalog branch.
+# Note: this is not the intended production use as it causes system dependent
+#       snapshots, but it is a minimally invasive solution for testing.
+setup() {
+    # setup up temporary local channel
+    export CHANNEL="$(mktemp -d)"
```

Comment:
Should you use `BATS_TEST_TMPDIR`? (I'm not honestly sure)

Reply:
Not sure if that is created per test or for the entire suite.